File: two-pointers/0125-valid-palindrome/valid-palindrome-solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s.size()<=1)
            return true;
        for(int i=0;i<s.size();i++)
        {
            if(s[i]>=48&&s[i]<=57)
                continue;
            if((s[i]>=65&&s[i]<=90)||(s[i]<=122&&s[i]>=97))
                continue;
            else s.erase(i--,1);
        }
        int mid=0;
        if(s.size()%2==0)
            mid=s.size()/2;
        else 
            mid=(s.size()-1)/2;
        int c=s.size();
        for(int i=0;i<c;i++)
            if(s[i]>=65&&s[i]<=90)
                s[i]=char(s[i]+32);
        for(int i=0;i<mid;i++)
        {
            if(s[i]!=s[c-i-1])
                return false;
        }
        return true;
    }
};

File: two-pointers/0125-valid-palindrome/valid-palindrome-solution.java
class Solution {
    public boolean isPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;

        while (left < right) {

            if (!Character.isLetterOrDigit(s.charAt(left))) {
                left++;
            }
            else if (!Character.isLetterOrDigit(s.charAt(right))) {
                right--;
            }
            // checking their lowercase to be equal here!
            else if (Character.toLowerCase(s.charAt(left)) ==
                     Character.toLowerCase(s.charAt(right))) {
                left++;
                right--;
            }
            else {
                return false;
            }
        }
        return true;
    }
}

